packages/core/src/utils/parseInvitation.ts

```diff
@@ -102,9 +105,33 @@ export const parseInvitationShortUrl = async (
   if (parsedUrl['oob']) {
     const outOfBandInvitation = OutOfBandInvitation.fromUrl(invitationUrl)
     return outOfBandInvitation
-  } else if (parsedUrl['c_i'] || parsedUrl['d_m']) {
+  } else if (parsedUrl['c_i']) {
     const invitation = ConnectionInvitationMessage.fromUrl(invitationUrl)
     return convertToNewInvitation(invitation)
+  }
+  // Legacy connectionless invitation
+  else if (parsedUrl['d_m']) {
+    const messageJson = JsonEncoder.fromBase64(parsedUrl['d_m'] as string)
+    const agentMessage = JsonTransformer.fromJSON(messageJson, AgentMessage)
+
+    // ~service is required for legacy connectionless invitations
+    if (!agentMessage.service) {
+      throw new AriesFrameworkError('Invalid legacy connectionless invitation url. Missing ~service decorator.')
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Maybe add a small comment that this is done to strip the `~service` from the object.